src/pythia/src/jobs/publisher.rs

```diff
@@ -0,0 +1,142 @@
+use std::{time::Duration, vec};
+
+use anyhow::{Context, Result};
+use ic_cdk::export::candid::Nat;
+use ic_cdk_timers::set_timer;
+use ic_dl_utils::retry_until_success;
+
+use futures::future::join_all;
+
+use super::withdraw;
+use crate::{
+    clone_with_state, log,
+    types::{
+        balance::Balances,
+        errors::PythiaError,
+        subscription::{Subscription, Subscriptions},
+        timer::Timer,
+    },
+    utils::{
+        abi, address, canister,
+        multicall::{multicall, Call},
+        nat, web3,
+    },
+};
+
+pub fn execute() {
+    ic_cdk::spawn(async {
+        if let Err(e) = _execute().await {
+            log!("error while executing publisher job: {e:?}");
+        }
+    })
+}
+
+async fn _execute() -> Result<()> {
+    log!("[PUBLISHER] publisher job started");
+    Timer::activate().context(PythiaError::UnableToActivateTimer)?;
```

Comment:
I really don't understand why you are using `anyhow::Context` and `thiserror` enum variants as messages for context as it really makes no sense at all